x-pack/legacy/plugins/siem/public/components/header_global/index.test.tsx

```diff
@@ -9,6 +9,8 @@ import React from 'react';
 
 import '../../mock/match_media';
 import { HeaderGlobal } from './index';
+// eslint-disable-next-line @kbn/eslint/module_migration
+import routeData from 'react-router';
```

Comment:
Optional. You can remove this pattern and use the suggested mocking patterns like so:

```ts
jest.mock('react-router-dom', () => ({
  useLocation: () => ({
    pathname: '/app/siem#/hosts/allHosts',
    hash: '',
    search: '',
    state: '',
  }),
  withRouter: () => jest.fn(),
}));
```

Then delete your mockHostsLocation below and:

```ts
jest.spyOn(routeData, 'useLocation').mockReturnValue(mockHostsLocation);
```

That way you don't have a linter issue and we can begin removing/stopping this tech debt pattern.

Just tired it out locally and it works in my env.

Reply:
Thank you! I was following suit from a Case test and was wondering if there was a way around this. 🙂